GlotPress/GlotPress-WP
# Add new PHP format

## What?


This adds a new `.php` _output_ format for translations.

> [!NOTE]  
>  Import is not yet implemented. It requires something like PHP-Parser to do that safely, as you don't want to include/require a random user-provided PHP file. This is best done in a separate PR.

## Why?


The WordPress core performance team is working on a [Performant Translations feature plugin](https://wordpress.org/plugins/performant-translations/) that uses a new approach to handle translation files in WordPress, making localization blazing fast. This plugin helps to make localized WordPress sites faster by replacing the traditional MO translation files with PHP files, which are much faster to parse. 

Learn more at https://make.wordpress.org/core/2023/09/05/call-for-testing-performant-translations/

Ideally, these PHP files are served directly from translate.wordpress.org, which is why support needs to be added to GlotPress first.

See https://meta.trac.wordpress.org/ticket/7296 for enabling that on the dotorg side of things.

## How?


Translations are exported using an optimized version of `var_export()` to use short array notation. A polyfill for `array_is_list` is included. Both are used to keep the file size as small as possible.

See https://make.wordpress.org/core/2023/09/05/call-for-testing-performant-translations/ and https://github.com/swissspidy/performant-translations

## Testing Instructions





1. Export translations as PHP
2. Check the syntax to see it's valid
3. Include/require that file in PHP, ideally in WP using that feature plugin.

## Screenshots or screencast
